atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServiceImpl.java

```diff
@@ -2435,6 +2436,8 @@ public void cleanUpSchemaMutationLockTablesState() throws Exception {
                 results.add(future.get());
             }
             return results;
+        } catch (ExecutionException e) {
+            throw Throwables.throwUncheckedException(e.getCause());
```

Comment:
Seems like there are other places in this class (e.g. `getRowsForSingleHost`, `getRowsColumnRangeForSingleHost`, `runAllTasksCancelOnFailure` above on line 2424) that may also want to unwrap to get the more streamlined cause, so might be worth adding something like the following and doing `%s/Throwables.throwUncheckedException/Throwables.unwrapAndThrowUncheckedException/g` in here:

```
    public static RuntimeException unwrapAndThrowUncheckedException(Throwable e) {
        if (e instanceof ExecutionException || e instanceof InvocationTargetException) {
            throw Throwables.throwUncheckedException(e.getCause());
        }
        throw Throwables.throwUncheckedException(e);
    }
```

Reply:
Most of the other places dont use the executor (line 2424 just runs the task), but I see how this can be an oversight going further, happy to add the check at other places too.